### ISSUE TITLE
Implement Rich Progress Bars for CLI Commands in mlstacks

### ISSUE THREAD

**Issue by strickvl** [zenml-io/mlstacks#125]
## Open Source Contributors Welcomed!
Please comment below if you would like to work on this issue!

### Contact Details [Optional]

support@zenml.io

### What happened?
Currently, the mlstacks CLI lacks visual feedback in the form of progress bars or spinners when executing commands like `mlstacks deploy ...` or `mlstacks destroy ...`. This can leave users uncertain about the progress or status of these commands.

### Task Description
Integrate [Rich progress bars](https://rich.readthedocs.io/en/stable/progress.html) into the `mlstacks` CLI to provide users with real-time visual feedback when executing commands. The goal is to enhance the user experience by making the interface more interactive and informative without overwhelming the user with excessive visuals.

### Expected Outcome
- Smooth and visually appealing progress bars or spinners should be displayed during the execution of `mlstacks deploy ...` and `mlstacks destroy ...` commands.
- The progress indicators should accurately represent the status and progression of the ongoing tasks.
- The implementation should not significantly impact the performance of the commands.

### Steps to Implement
- Explore [the Rich library](https://rich.readthedocs.io/en/latest/) and identify suitable progress bar or spinner styles for integration.
- Implement the progress bars/spinners in the `mlstacks` CLI, particularly for the deploy and destroy commands.
- Ensure that the progress indicators are accurately tied to the actual process milestones.
- Test the implementation across different platforms to ensure consistent behavior and appearance.
- (optionally) Update documentation to mention these new UI features for user awareness.

### Additional Context
This enhancement aims to improve the overall user experience of the `mlstacks` CLI, making it more engaging and informative, which is especially helpful in managing expectations during longer-running processes.

### Code of Conduct
- [ ] I agree to follow this project's Code of Conduct

**Comment by code-master-ajay** [comment]
I would like to take this up.

**Comment by strickvl** [comment]
Great. Let me know if you have any questions about this, and def take a look at [our sister repo](https://github.com/zenml-io/zenml/tree/main/src/zenml/cli) where we do similar things with progress bars etc.

**Comment by code-master-ajay** [comment]
Sure, thanks for letting me contribute.

**Comment by strickvl** [comment]
@code-master-ajay any progress on this or questions about the implementation / approach you want to take?

**Comment by code-master-ajay** [comment]
Hi @strickvl, i looked up rich library and tried to use it with the terraform deploy function, but i am not able to get the terraform output in the code to be able to show the progress.

please let me know if i am doing something wrong, or should i try any other approach.
Sorry for the delayed update.

**Comment by strickvl** [comment]
So I think it might be as simple as emulating how we handle it here: https://github.com/zenml-io/zenml/blob/cdd14528d674e0cafe00173755e556ea00c6343c/src/zenml/cli/integration.py#L274.

I.e. using the context handler around the deployment call. But in any case would be useful to see specifically what you tried as it's hard to give feedback without more information.

**Comment by strickvl** [comment]
Yeah no need to make it so complicated, I think :) The `console.status()` method shows a spinner, so I think that's basically what we're looking for.

**Comment by code-master-ajay** [comment]
Sure @strickvl , I will do that . Thanks for the guidance :)
